### PR TITLE
Added Definitions for jSuite and SuiteScript

### PR DESCRIPTION
**Pull request by darrenhillconsulting**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: suitescript/suitescript.d.ts [hunk 1]
```diff
@@ -1234,7 +1236,7 @@ declare function nlapiGetLineItemDateTimeValue(type:string, fldnam:string, linen
  * @param linenum
  * @param value
  */
-declare function nlapiSetLineItemValue(type:string, fldnam:string, linenum:any, value:string):void;
+declare function nlapiSetLineItemValue(type:string, fldnam:string, linenum:any, value:any):void;
```
Comment (RyanCavanaugh) [inline suitescript/suitescript.d.ts]: This is documented as taking a `string`, why change to `any` ?

File: suitescript/suitescript.d.ts [hunk 2]
```diff
@@ -2966,7 +3017,7 @@ declare interface nlobjRecord {
      * @since 2009.2
      * @param group
      */
-    getLineItemCount(group:string): void;
+    getLineItemCount(group:string): any;
```
Comment (RyanCavanaugh) [inline suitescript/suitescript.d.ts]: Not `: number` ?

Comment (darrenhillconsulting) [inline suitescript/suitescript.d.ts]: It is with my 6yrs experience with the API that I know this function return
string as well as in, but also null and undefined.

You'll notice the bulk of my updates are correcting these function
signatures, not really too many ADDs

Comment (RyanCavanaugh) [inline suitescript/suitescript.d.ts]: so `: string | number` ?

File: suitescript/suitescript.d.ts (hunk 1 above)
Comment (darrenhillconsulting) [inline suitescript/suitescript.d.ts]: Much like my last comment, based on history.  But, I'll take your approach and change to string|number

**Pull request merged**